src/main/java/de/tum/in/www1/artemis/domain/BuildPlan.java

```diff
@@ -0,0 +1,37 @@
+package de.tum.in.www1.artemis.domain;
+
+import javax.annotation.Nullable;
+import javax.persistence.Column;
+import javax.persistence.Entity;
+import javax.persistence.Table;
+import javax.persistence.UniqueConstraint;
+import javax.validation.constraints.Size;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+
+@Entity
+@Table(name = "build_plan", uniqueConstraints = { @UniqueConstraint(columnNames = { "build_plan" }) })
+@JsonInclude(JsonInclude.Include.NON_EMPTY)
+public class BuildPlan extends DomainObject {
+
+    @Size(max = 10_000)
+    @Nullable
+    @Column(name = "build_plan", table = "build_plan", length = 10_000)
+    private String buildPlan;
```

Comment:
This should probably be a "name"? `buildPlan.getBuildPlan()` sounds off to me.

Reply:
Well, the name of the class and attribute are maybe a little bit complicated to understand, but I think it is ok how it is right now.
During the system design, we have thought a lot about the database model changes and we decided to do it this way. During my thesis presentation, there were no objections made, so I have implemented it like it is now.
I think, it would be nice if the implementation is consistent with my thesis which I already submitted (so I cannot change the names there).

Anyways, this is just a temporary solution as we have two future work aspects, which probably lead to a change regarding this aspect:
- First of all, the build plans must be changed in MySQL (or Postgres) right now. Therefore, we plan to provide an editor in Artemis to do so.
- On a larger scale, it is not intuitive to store the whole build plan. In our "dream solution" we only store the commands which have to be executed, and Artemis transforms the sequence of commands in a build plan suitable to the CI system. All build plans in Artemis are cross-system compatible and just built to the "real" built plan whenever needed.
Then the name is irrelevant anyhow.